src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -15,58 +19,80 @@
 import java.nio.file.Files;
 import java.util.Arrays;
 import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 
 /**
  * Utilities for dealing with google buckets.
  */
 public final class BucketUtils {
-    public static final String GCS_SCHEME = "gs";
-    public static final String GCS_PREFIX = GCS_SCHEME + "://";
+    public static final String GCS_PREFIX = GoogleCloudStorageFileSystem.SCHEME + "://";
+    public static final String HTTP_PREFIX = HttpFileSystemProvider.SCHEME + "://";
+    public static final String HTTPS_PREFIX = HttpsFileSystemProvider.SCHEME +"://";
     public static final String HDFS_PREFIX = "hdfs://";
 
     // slashes omitted since hdfs paths seem to only have 1 slash which would be weirder to include than no slashes
     public static final String FILE_PREFIX = "file:";
 
-    public static final Logger logger = LogManager.getLogger("org.broadinstitute.hellbender.utils.gcs");
-
     private BucketUtils(){} //private so that no one will instantiate this class
 
-    public static boolean isCloudStorageUrl(final String path) {
+    /**
+     * @param path path to inspect
+     * @return true if this path represents a gcs location
+     */
+    public static boolean isGcsUrl(final String path) {
         Utils.nonNull(path);
         return path.startsWith(GCS_PREFIX);
     }
 
     /**
-     * Return true if this {@code GATKPathSpecifier} represents a gcs URI.
      * @param pathSpec specifier to inspect
-     * @return true if this {@code GATKPathSpecifier} represents a gcs URI.
+     * @return true if this {@code GATKPathSpecifier} represents a remote storage system which may benefit from prefetching (gcs or http(s))
      */
-    public static boolean isCloudStorageUrl(final GATKPathSpecifier pathSpec) {
+    public static boolean isEligibleForPrefetching(final GATKPathSpecifier pathSpec) {
         Utils.nonNull(pathSpec);
-        return pathSpec.getScheme().equals(GCS_SCHEME);
+        return isEligibleForPrefetching(pathSpec.getScheme());
+     }
+
+    /**
+     * @param path path to inspect
+     * @return true if this {@code Path} represents a remote storage system which may benefit from prefetching (gcs or http(s))
+     */
+    public static boolean isEligibleForPrefetching(final java.nio.file.Path path) {
+        Utils.nonNull(path);
+        return isEligibleForPrefetching(path.toUri().getScheme());
     }
 
-    public static boolean isCloudStorageUrl(final java.nio.file.Path path) {
-        // the initial "" protects us against a null scheme
-        return ("" + path.toUri().getScheme() + "://").equals(GCS_PREFIX);
+    private static boolean isEligibleForPrefetching(final String scheme){
+        return scheme != null
+                && (scheme.equals(GoogleCloudStorageFileSystem.SCHEME)
+                || scheme.equals(HttpFileSystemProvider.SCHEME)
+                || scheme.equals(HttpsFileSystemProvider.SCHEME));
```

Comment:
Would be better if you could call the `isGcsUrl()`/`isHttpUrl()` methods from here, but no big deal (@cmnbroad is probably going to refactor all of this code anyway as part of the `GATKPathSpecifier` migration..). More importantly: have you actually tested the prefetcher with HTTP/HTTPs inputs to make sure that it works properly?

Reply:
Yes, it's used in the the PrintReads http tests.  Going to leave it how it is because I factored it to use the scheme rather than the whole url at the moment.